app/models/piece.rb

```diff
@@ -86,6 +79,15 @@ def move_obstructed?(to_row, to_col)
     false
   end
 
+  def self_check?(to_row, to_col)
+    from_row = row
+    from_col = col
+    update(row: to_row, col: to_col)
```

Comment:
Are there any suggestions on how to refactor? `self_check?` updates the board because it relies on `in_check`, which relies `valid_move?`, which relies on the state of the board. The only way I can think of refactoring is to 'clone' the board state onto an instance of Board and write a version of `valid_move?` that works on Board instances.